src/dotnet/commands/dotnet-restore3/Restore3CommandApp.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Restore3
+{
+    public class Restore3CommandApp
+    {
+        public static int Execute(string[] args)
+        {
+            CommandLineApplication cmd = new CommandLineApplication
+            {
+                Name = "restore3",
+                FullName = "restore3",
+                Description = "restore for msbuild"
+            };
+
+            cmd.HelpOption("-h|--help");
+
+            var argRoot = cmd.Argument(
```

Comment:
That works today because the msbuild args start with ``/`` and this parser looks for ``-``. Unless you can also pass ``-`` to msbuild, is that possible?

I don't see a way to validate unexpected args in msbuild, so if we set this to false there wouldn't be any help.